### PR TITLE
Filter Auth methods by name or type

### PR DESCRIPTION
**Pull request by Monkeychip**
Pulled from the previous work done [here](https://github.com/hashicorp/vault/pull/20481/files). This adds the same functionality to the authentication list view.

Inspired by [this GH Issue](https://github.com/hashicorp/vault/issues/19001).

Movie of the filtering in action:

https://github.com/hashicorp/vault/assets/6618863/565ee026-15bc-45b2-bfda-641a5da09879

### CODE REVIEW

File: ui/tests/acceptance/secrets/backend/engines-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/engines-test.js:1]: Noticed I needed to clean this up from my last test writing for the secret engine filtering.

File: ui/app/controllers/vault/cluster/access/methods.js [hunk 1]
```diff
@@ -4,22 +4,72 @@
...
+  pageFilter = null;
+  filter = null;
+
+  get authMethodList() {
+    // return an options list to filter by engine type, ex: 'kv'
+    if (this.selectedAuthType) {
+      // check first if the user has also filtered by name.
+      if (this.selectedAuthName) {
+        return this.model.filter((method) => this.selectedAuthName === method.id);
+      }
```
Comment (zofskeez) [inline ui/app/controllers/vault/cluster/access/methods.js:31]: I'm curious if we should be filtering by both type and name if they are both set? I think there could be a case where the same name is used for different types.

File: ui/app/routes/vault/cluster/access/methods.js
Comment (zofskeez) [inline ui/app/routes/vault/cluster/access/methods.js:1]: ✨

**Review by zofskeez — COMMENTED** [review]
This all looks good but I'm interested in the sorting by type and name logic. If I'm off base with my thinking around that then I'll go ahead and approve.

File: ui/app/controllers/vault/cluster/access/methods.js (hunk 1 above)
Comment (Monkeychip) [reply ui/app/controllers/vault/cluster/access/methods.js:31]: For secret engines you can't have the same name as used by another secret engine. We do some validation work on that and say as much. And if they passed validation, then the API would return the same error.

But great question for auth methods. I had to check if this was the same. Turns out it is. We don't do any validation but we do return the API error.
image

So names are individualized across all auth methods. Which I believes answers your question?

Comment (zofskeez) [reply ui/app/controllers/vault/cluster/access/methods.js:31]: Ok good to know thanks! Maybe nice to add a comment in there stating as much so it's clear why type is ignored if name is set.

File: ui/app/controllers/vault/cluster/access/methods.js [hunk 2]
```diff
@@ -4,22 +4,72 @@
...
+
+  @action
+  filterAuthType([type]) {
+    this.selectedAuthType = type;
+  }
+
+  @action
+  filterAuthName([name]) {
+    this.selectedAuthName = name;
+  }
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/controllers/vault/cluster/access/methods.js:69]: Good thought. The only issue is that the sorted list that is passed around is pretty unique. For KV we need to combine and filter two lists (supported and unsupported) and here we're just using the this.model and filtering there. With KV the unique list situation meant a couple of extra lines of code not seen here. But it's a great consideration. I'll think about this with the KV work as the new models may allow us to standardize the list.

File: ui/app/routes/vault/cluster/access/methods.js
Comment (Monkeychip) [reply ui/app/routes/vault/cluster/access/methods.js:1]: couldn't help myself. I opened the file and had to glimmerize.

**Review by zofskeez — APPROVED** [review]

**Pull request merged**